src/modules/aws_ecs_fargate@0.0.1/entity/aws_container_definition.ts

```diff
@@ -19,7 +19,10 @@ export enum TransportProtocol {
   UDP = "udp"
 }
 
-@Check(`"docker_image" is not null or "repository_id" is not null  or "public_repository_id" is not null`)
+// `image` > `repository` > `publicRepository`
+// `digest` > `tag` > null
+@Check(`("image" is null and ("repository_id" is not null or "public_repository_id" is not null)) or "image" is not null`)
+@Check(`("image" is null and ("tag" is not null or "digest" is not null)) or "image" is not null`)
```

Comment:
With these as two separate checks won't it cause a failure outright on `"image" is null`? Because the first one could pass but the second would not and vice versa?

Reply:
Hmm what case do you have in mind? I do not see it fail? 
If `image is null` It is checking that an ECR exists, if an ECR exists a `tag` or `digest` have to exists.
If `image is not null` both pass

Reply:
Oh there's a case where someone define the container with an ECR but no `tag` or `digest` and it would fail. Let me review that!

Reply:
So, I thought you mentioned some sort of "default tag" yesterday? Is that not the case?

The other failure direction I thought I saw with the first check I now see wouldn't happen, though.

Reply:
Ok so it would be just remove the second check. It is not really necessary, we keep `tag` and `digest` as nullable/optional values.

Reply:
Wait, keep it, but just make it such that you can't have both at the same time, but you can have neither?

Reply:
```suggestion
@Check(`("image" is null and ("repository_id" is not null or "public_repository_id" is not null)) or "image" is not null`)
@Check(`("image" is null and (("tag" is not null and "digest" is null) or ("tag" is null and "digest" is not null) or ("tag" is null and "digest" is null)) or "image" is not null`)
```

Reply:
Oh well, I was being permissive and just have this around (and eventually added in the docs?) `digest` > `tag` > null meaning that if both defined, `digest` will be taken into account first since it is more specific. Same for ECR and `image` fields being defined at the same time, `image` > `repository` > `publicRepository`. I know this is opinionated, we can defined as we want.

Reply:
> Suggested change 
> @Check(`("image" is null and ("repository_id" is not null or "public_repository_id" is not null)) or "image" is not null`)
> @Check(`("image" is null and ("tag" is not null or "digest" is not null)) or "image" is not null`)
> @Check(`("image" is null and ("repository_id" is not null or "public_repository_id" is not null)) or "image" is not null`)
> @Check(`("image" is null and (("tag" is not null and "digest" is null) or ("tag" is null and "digest" is not null) or ("tag" is null and "digest" is null)) or "image" is not null`)

If we want this, the `image` check still need to go, because you can have a ECR uri only apparently, with no tag or digest. It would be something like: 
```suggestion
@Check(`("image" is null and ("repository_id" is not null or "public_repository_id" is not null)) or "image" is not null`)
@Check(`("tag" is not null and "digest" is null) or ("tag" is null and "digest" is not null)`)
```

Reply:
That's missing the path where they are both null, though?

Reply:
indeed 
```suggestion
@Check(`("image" is null and ("repository_id" is not null or "public_repository_id" is not null)) or "image" is not null`)
@Check(`("tag" is null and "digest" is null) or ("tag" is not null and "digest" is null) or ("tag" is null and "digest" is not null)`)
```